cmd/src/search.go

```diff
@@ -419,6 +420,31 @@ type highlight struct {
 	length    int // the 1-indexed length of the highlight, in characters.
 }
 
+// applyHighlightsForFile expects highlight information that is
+// relative to the whole file, and can add lines of context around the
+// highlighted range. It makes no assumptions about the preview field in
+// LineMatches (the preview field is not used).
+func applyHighlightsForFile(fileContent *string, highlights []highlight) string {
```

Comment:
```suggestion
func applyHighlightsForFile(fileContent string, highlights []highlight) string {
```

It is weird for this to be a pointer (Go's `string` is already a pointer).